concept/validation/schema-validation.feature

```diff
@@ -0,0 +1,733 @@
+        rel11 sub rel0, relates role11 as role0;
+
+        ent0 sub entity, plays rel0:role0;
+        ent1 sub ent0, plays rel10:role10 as role0;
+        ent20 sub entity, plays rel0:role0;               # plays will be hidden under ent1
+        ent21 sub ent0, plays rel11:role11 as role0;      # Overridden will be hidden under ent1
+        ent22 sub ent0, plays rel10:role10; # as role0;   # Will be redundant under ent1
+    """
+    Given transaction commits
+
+    When session opens transaction of type: write
+    Then entity(ent20) set supertype: ent1; throws exception
+
+    When session opens transaction of type: write
+    When entity(ent21) set supertype: ent1
+    Then transaction commits; throws exception
+
+    When session opens transaction of type: write
+    When entity(ent22) set supertype: ent1
+    Then transaction commits; throws exception
+
+
+  Scenario: A concrete type must override any ownerships of abstract attributes it inherits.
+    Given put attribute type: attr00, with value type: string
+    Given attribute(attr00) set abstract: true
+    Given put attribute type: attr10, with value type: string
+    Given attribute(attr10) set supertype: attr00
+    Given put attribute type: attr01, with value type: string
+    Given attribute(attr01) set abstract: true
+    Given put attribute type: attr11, with value type: string
+    Given attribute(attr11) set supertype: attr01
+    Given put entity type: ent00
+    Given entity(ent00) set abstract: true
+    Given entity(ent00) set owns attribute type: attr00
+    Given put entity type: ent01
+    Given entity(ent01) set abstract: true
+    Given entity(ent01) set owns attribute type: attr01
+    When put entity type: ent1
+    Given transaction commits
+
+    # inherits abstract ownership but does not override
+    When session opens transaction of type: write
+    When entity(ent1) set supertype: ent00
+    Then transaction commits; throws exception
+
+    # declares concrete ownership with a subtype but is missing override clause
+    When session opens transaction of type: write
+    When entity(ent1) set supertype: ent00
+    When entity(ent1) set owns attribute type: attr10
+    Then transaction commits; throws exception
+
+    When session opens transaction of type: write
+    When entity(ent1) set supertype: ent00
+    When entity(ent1) set owns attribute type: attr10 as attr00
+    Then transaction commits
+
+    When session opens transaction of type: write
+    When entity(ent1) set supertype: ent00
+    When entity(ent1) set owns attribute type: attr10 as attr00
+    Then transaction commits
+
+    When session opens transaction of type: write
+    When entity(ent1) set owns attribute type: attr11
+    Then transaction commits
+
+    When session opens transaction of type: write
+    When entity(ent1) set supertype: ent01
+    Then transaction commits; throws exception
+
+
+  Scenario: Types which are referenced in rules may not be renamed
```

Comment:
Just flagging this as an interesting error that's always been caught at commit time.